tests/test_args.cpp

```
@@ -0,0 +1,70 @@
+// truncated the step to zero and the loop free-ran, 1e-30 overflowed the step
+// duration and hung shutdown, and nan passed the bounds check outright.
+TEST_CASE("degenerate physics rates are rejected") {
+  for (const char* rate :
+       {"inf", "-inf", "nan", "1e300", "1e-30", "0", "-5", "2e6"}) {
+    CAPTURE(rate);
+    CHECK_THROWS_AS(parse({"volasim", "--physics-hz", rate}),
+                    std::runtime_error);
+  }
+}
+
+TEST_CASE("malformed physics rates are rejected") {
```

Clang (14.0.6)

[error] function 'DOCTEST_ANON_FUNC_13' has cognitive complexity of 26 (threshold 25)

(readability-function-cognitive-complexity,-warnings-as-errors)